sites/website/src/docs/apps-and-experiences/dependency-injection.md

```diff
@@ -0,0 +1,181 @@
+---
+id: dependency-injection
+title: Dependency Injection
+sidebar_label: Dependency Injection
+custom_edit_url: https://github.com/microsoft/fast/edit/master/sites/website/src/docs/apps-and-experiences/dependency-injection.md
+---
+
+An important principle of object oriented programming (OOP) is to favor a compositional approach to systems over an inheritance-based approach. In such systems complex problems are broken down into small, single-purpose objects that collaborate with one another. However, this approach is not a panacea, and introduces its own set of challenges:
+
+- How do we instantiate a set of collaborating objects, particularly when there is a complex arrangement of dependencies between the objects?
+- How do we avoid tight coupling between an object and the implementation details of its dependencies?
+- How do we manage memory and control the lifetimes of objects in such a system?
+- How does our user interface, often controlled by a different engine, gain access to our composed system capabilities?
+
+To address these challenges, FAST introduces the concept of a *dependency injection container*. A DI Container is a sub-system with the responsibilities of understanding dependency relationships, constructing objects with their dependencies, delivering dependencies to components, and managing lifetimes.
+
+## Creating a DI Container
+
+DI Containers can exist in a hierarchy, allowing child containers to override the dependencies of the parent, but typically there is one root container in which all the system's services are registered. In a typical FAST application, you will want this container to be associated with the `document.body` so that all UI component children can gain access to its capabilities. Here's how you would create that root container:
+
+```ts
+const container = DI.getOrCreateDOMContainer();
+```
+
+You'll want to create and configure your root container as early as possible in your application lifecycle, typically in your application's entry point module. If you are using FAST's Design System features or its components, we've integrated them with DI, so that you can figure everything with a unified API. Instead of calling `DI.getOrCreateDOMContainer()` you can simply import the Design System Provider function and use that. Here's some code that you may have seen in other parts of our documentation, that does just that:
```

Comment:
```suggestion
You'll want to create and configure your root container as early as possible in your application lifecycle, typically in your application's entry point module. If you are using FAST's Design System features or its components, we've integrated them with DI, so that you can configure everything with a unified API. Instead of calling `DI.getOrCreateDOMContainer()` you can simply import the Design System Provider function and use that. Here's some code that you may have seen in other parts of our documentation, that does just that:
```